figo/figo.py

```diff
@@ -1,290 +1,192 @@
+from .version import __version__
 
+load_dotenv()
 logger = logging.getLogger(__name__)
 
 API_ENDPOINT = os.getenv("FIGO_API_ENDPOINT")
 
-ERROR_MESSAGES = {
-    400: {
-        'message': "bad request",
-        'description': "Bad request",
-        'code': 90000,
-    },
-    401: {
-        'message': "unauthorized",
-        'description': "Missing, invalid or expired access token.",
-        'code': 90000,
-    },
-    403: {
-        'message': "forbidden",
-        'description': "Insufficient permission.",
-        'code': 90000,
-    },
-    404: {
-        'message': "not_found",
-        'description': "Not found.",
-        'code': 90000,
-    },
-    405: {
-        'message': "method_not_allowed",
-        'description': "Unexpected request method.",
-        'code': 90000,
-    },
-    423: {
-        'message': 'resource_locked',
-        'description': 'Resource locked',
-        'code': 1008,
-    },
-    503: {
-        'message': "service_unavailable",
-        'description': "Exceeded rate limit.",
-        'code': 90000,
-    },
-    504: {
-        'message': 'upstream_request_timeout',
-        'description': 'Upstream request timeout.',
-        'code': 90000,
-    },
-}
-
-
-def getAccountId(account_or_account_id):
-    if account_or_account_id is None:
-        return None
-    elif isinstance(account_or_account_id, Account):
-        return account_or_account_id.account_id
-    else:
-        return account_or_account_id
-
-
-def filterKeys(object, allowed_keys):
-    if object is None or object == {}:
-        return {}
-    else:
-        keys = [key for key in object.keys() if key in allowed_keys]
-        return dict(zip(keys, [object[key] for key in keys]))
-
-
-def filterNone(object):
-    return {k: v for k, v in object.items() if v is not None}
-
-
-class FigoObject(object):
+
+class FigoObject:
     """A FigoObject has the ability to communicate with the Figo API."""
 
     def __init__(self, api_endpoint=API_ENDPOINT, language=None):
         """Create a FigoObject instance.
 
         Args:
-            api_endpoint (str) - base URI of the server to call
-            language (str) - language for HTTP request header
+            api_endpoint (str): base URI of the server to call
+            language (str): language for HTTP request header
         """
         self.headers = {
-            'Accept': 'application/json',
-            'Content-Type': 'application/json',
-            'User-Agent': "python_figo/{0}".format(__version__),
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "User-Agent": f"python_figo/{__version__}",
         }
         self.language = language
         self.api_endpoint = api_endpoint
 
-    def _request_api(self, path, data=None, method="GET", raise_exception=False):
+    @staticmethod
+    def _has_error(response):
+        return "error" in response and response["error"]
```

Comment:
```suggestion
        return bool(response.get("error"))
```

Reply:
This change broke test:

```python
---------------------------------------------------------------------------------------------------------------------- Captured stderr call -----------------------------------------------------------------------------------------------------------------------
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): staging.figo.me:443
DEBUG:urllib3.connectionpool:https://staging.figo.me:443 "GET /rest/accesses/X677030772078018561.1/syncs/aac860f0-93d3-4f85-90b3-2bb0b87711be/challenges HTTP/1.1" 200 189
DEBUG:figo.figo:GET 'https://staging.figo.me/rest/accesses/X677030772078018561.1/syncs/aac860f0-93d3-4f85-90b3-2bb0b87711be/challenges' result with status 200 and text: [{"id": "9fc8be3c-f1c3-4d79-9696-6728e3312187.1", "created_at": "2021-07-19T08:36:51.348191+00:00", "type": "EMBEDDED", "label": "", "format": "TEXT", "data": "Enter valid OTP to log in."}]
------------------------------------------------------------------------------------------------------------------------ Captured log call ------------------------------------------------------------------------------------------------------------------------
DEBUG    urllib3.connectionpool:connectionpool.py:975 Starting new HTTPS connection (1): staging.figo.me:443
DEBUG    urllib3.connectionpool:connectionpool.py:461 https://staging.figo.me:443 "GET /rest/accesses/X677030772078018561.1/syncs/aac860f0-93d3-4f85-90b3-2bb0b87711be/challenges HTTP/1.1" 200 189
DEBUG    figo.figo:figo.py:90 GET 'https://staging.figo.me/rest/accesses/X677030772078018561.1/syncs/aac860f0-93d3-4f85-90b3-2bb0b87711be/challenges' result with status 200 and text: [{"id": "9fc8be3c-f1c3-4d79-9696-6728e3312187.1", "created_at": "2021-07-19T08:36:51.348191+00:00", "type": "EMBEDDED", "label": "", "format": "TEXT", "data": "Enter valid OTP to log in."}]
===================================================================================================================== short test summary info =====================================================================================================================
FAILED tests/test_integrations.py::test_get_accesses - AttributeError: 'list' object has no attribute 'get'
FAILED tests/test_integrations.py::test_get_synchronization_challenges - AttributeError: 'list' object has no attribute 'get'
============================================================================================================= 2 failed, 53 passed in 63.24s (0:01:03) =============================================================================================================
```

I will revert this commit.

Reply:
Instead of revert commit I use condition:

```python
response = response[0] if isinstance(response, list) else response
```